pytorch_lightning/utilities/auto_restart.py

```diff
@@ -168,6 +172,16 @@ def update(self, generator_name: Optional[str], new_state: IteratorState) -> Non
         state[latest_worker_id] = new_state
         self.latest_worker_id = latest_worker_id
 
+    @property
+    def sampler_states(self) -> Dict[int, Any]:
+        # TODO: add docs
+        # TODO: double check the index here:
+        return {0: self.state[k].sampler_state[0] for k in self.state.keys()}
```

Comment:
@tchaton is the index 0 correct here?

Reply:
I am not entirely sure tbh. We need some test with num_workers > 0.

Reply:
we only use the sampler_states property in our test, not in the lightning integration. the test works with num_workers and if I change it here, it breaks.